package/MDAnalysis/analysis/helix_analysis.py

```diff
@@ -0,0 +1,515 @@
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+HELANAL --- analysis of protein helices
+=======================================
+
+:Author: Lily Wang
+:Year: 2020
+:Copyright: GNU Public License v3
+
+.. versionadded:: 1.0.0
+
+This module contains code to analyse protein helices using the 
+`HELANAL <http://nucleix.mbu.iisc.ernet.in/helanal/helanal/helanal.html>`_ algorithm
+([Bansal2000]_ , [Sugeta1967]_ ).
+
+`HELANAL <http://nucleix.mbu.iisc.ernet.in/helanal/helanal/helanal.html>`_ 
+quantifies the geometry of helices in proteins on the basis of their Cα
+atoms. It can determine local structural features such as the local helical twist and
+rise, virtual torsion angle, local helix origins and bending angles between
+successive local helix axes. 
+
+Example use
+-----------
+
+You can pass in a single selection::
+
+    import MDAnalysis as mda
+    from MDAnalysis.tests.datafiles import PSF, DCD
+    from MDAnalysis.analysis import helix_analysis as hel
+    u = mda.Universe(PSF, DCD)
+    helanal = hel.HELANAL(u, select='name CA and resnum 161-187')
+    helanal.run()
+
+    print(helanal.summary)
+
+Alternatively, you can analyse several helices at once by passing 
+in multiple selection strings::
+
+    helanal2 = hel.HELANAL(u, select=('name CA and resnum 100-160',
+                                      'name CA and resnum 200-230'))
+
+"""
+from __future__ import division, absolute_import
+
+import warnings
+import numpy as np
+
+import MDAnalysis as mda
+from ..lib import util, mdamath
+from .base import AnalysisBase
+
+
+def pdot(a, b):
+    """Pairwise dot product.
+
+    ``a`` must be the same shape as ``b``.
+
+    Parameters
+    ----------
+    a: :class:`numpy.ndarray` of shape (N, M)
+    b: :class:`numpy.ndarray` of shape (N, M)
+
+    Returns
+    -------
+    :class:`numpy.ndarray` of shape (N,)
+    """
+    return np.einsum('ij,ij->i', a, b)
+
+
+def pnorm(a):
+    """Euclidean norm of each vector in a matrix
+
+    Parameters
+    ----------
+    a: :class:`numpy.ndarray` of shape (N, M)
+
+    Returns
+    -------
+    :class:`numpy.ndarray` of shape (N,)
+    """
+    return pdot(a, a)**0.5
+
+
+def vector_of_best_fit(coordinates):
+    """Fit vector through the centered coordinates.
+
+    Parameters
+    ----------
+    coordinates: :class:`numpy.ndarray` of shape (N, 3)
+
+    Returns
+    -------
+    :class:`numpy.ndarray` of shape (3,)
+        Vector of best fit.
+    """
+    centered = coordinates - coordinates.mean(axis=0)
+    Mt_M = np.matmul(centered.T, centered)
+    u, s, vh = np.linalg.linalg.svd(Mt_M)
+    vector = vh[0]
+
+    # does vector face first residue?
+    angle = mdamath.angle(coordinates[0], vector)
```

Comment:
> I think this should be centered[0] rather than coordinates[0]

D'oh. Yes, you're right.

> Related thought: is there a reason to chose the direction of the helix vector to point towards the 1st residue? I don't remember if it's in any of the references, but at least to me it makes more sense intuitively for the helix axis to point towards the last residue (which would be the same direction as the local helix axes).

Each origin point would have its own local helix axis associated with it, so I don't understand if this matters.

I do agree with your intuition, it feels upside-down. However since it only affects whether the global tilt is obtuse or acute, I left it like that -- also because I peeked at how [these guys](https://github.com/davecao/biHelix/blob/00f6e3905b0171e604be2edc4c7a61096893b3ea/src/helanal.f90#L367-L374) did it and they also point at the first local origin point (that comment is inaccurate).

Reply:
Yeah, that's fair - no harm in keeping it as is! Either way, it might be worth clarifying in the docs that it's end-to-start in case anyone uses the `global_axes` output and the orientation matters